### ISSUE TITLE
Running tests in nodejs with jsdom

### ISSUE THREAD

**Issue by vloth** [lilactown/helix#136]
## Is it possible to run helix with jsdom?

It seems like helix requires **window** to be defined at compilation time, and shadow-clj doesn't support user defined javascript while bundling.
Running tests gives me:
```
SHADOW import error helix-jsdom/.shadow-cljs/builds/node-tests/dev/out/cljs-runtime/main.app.js

/helix-jsdom/.shadow-cljs/builds/node-tests/dev/out/cljs-runtime/helix/core.cljs:237
  (and (exists? (.-$$Signature$$ js/window))
       ^
ReferenceError: window is not defined
    at Object.helix$core$signature_BANG_ [as signature_BANG_] 
```

[I created a minimal reproducible project here](https://github.com/vloth/helix-jsdom).
If that's not supported, is that something you are planning to support in future?
(I don't know if that's relevant, but I did manage to make it work with [reagent](https://github.com/vloth/reagent-boilerplate)).
Is there anything I could do to help?

Thanks for creating and sharing helix!

**Comment by lilactown** [comment]
yeah that sounds like a good change. I've been bit by this bug before, would certainly accept a patch!

**Comment by vloth** [comment]
@rome-user thanks for investigating and submitting the patch!

I had to fix the jsdom setup a little bit, but from the tests I did, it seems to be working with the suggested fix!
To fix the setup, I extracted the component under test into another file (that's the version in the main branch).
I also pushed another version to `document-load-free` branch which just defers the document use and it's working well too.

Looking forward to see the patch applied, thank you again!
